### ISSUE TITLE
Create a new resource - feedback

### ISSUE THREAD

**Issue by viktorsmari** [reflow-project/weloop#77]
it would be valuable for you to pick up the options of the 'create a new resources' drop down menus from the deliverable 5.2, especially because we set-up those drop down menus for each use cases, such as the unit and the processes. 

Right now, I find it a bit complicated to know how to enter for instance : 

resource: doors
localisation : RE-STORE 93200 Saint Denis France
Action : stored and ready to be sell
provider : deconstruction site 
receiver : recovery center
unit : quantity
numerical value : 20

**Comment by sbocconi** [comment]
@viktorsmari as far as I understand Valueflows, the actions that you list are related to an event (EconomicEvent), not to a resource (EconomicResource). As far as I understand in Valueflows you should be able to define a resource and then apply to it an event, which could be use, consume, transfer, etc.

Or is this a particular interpretation made by Reflow of Valueflows?

**Comment by viktorsmari** [comment]
Correct @sbocconi, but I think in order to create the a `economicResource` from scratch, an `economicEvent` is needed, using an `action` of either `produce` or `transfer`

**Comment by viktorsmari** [comment]
Then when we have created a `economicResource`, we should see it under `My Inventory`

![2021-06-10_12-53-07](https://user-images.githubusercontent.com/1689020/121513200-dea7c800-c9ea-11eb-8faa-7c75f9a78d72.png)

... where we can **Perform event**s on it
![2021-06-10_12-53-24](https://user-images.githubusercontent.com/1689020/121513195-de0f3180-c9ea-11eb-9caa-cbeadb69aa52.png)

**Comment by sbocconi** [comment]
> Correct @sbocconi, but I think in order to create an `economicResource` from scratch, an `economicEvent` is needed, using an `action` of either `produce` or `transfer`

Any event has also a resource as input usually, so in my view resources are the starting point of a flow, such as for example copper to make wires, or seeds that create apple trees that create apples, etc.

**Comment by pral2a** [comment]
> Correct @sbocconi, but I think in order to create an economicResource from scratch, an economicEvent is needed, using an action of either produce or transfer

@sbocconi I can confirm @viktorsmari is right, to use the **Value Flows** paradigm properly to **create** or **update** any `economicResource` you need to use an `economicEvent` that defines a specific `action` (like the ones listed by @viktorsmari on the issue).  The `action` action property designates how it will affect an `economicResource` (or not).

**Comment by sbocconi** [comment]
@pral2a yes to act upon a resource you always need an event. But to **define** a resource you do not need an event, look for example at the examples at https://valueflo.ws/examples/ex-resource.html where resources are defined, and then used in events.

**Comment by viktorsmari** [comment]
@sbocconi do you know how to create the **first** resource without using events?

We used this example as a reference:
https://reflowos.dyne.org/docs/api_tour#conduct-an-economic-event-that-results-in-a-completely-new-resource

**Comment by sbocconi** [comment]
That is why I was asking whether it is a particular Reflow interpretation of Valueflows, so that in the Reflow version of Valueflows it is not possible, but it is possible in Valueflows as defined by https://valueflo.ws/

**Comment by pral2a** [comment]
> https://valueflo.ws/examples/ex-resource.html 

I don't think it answers the questions as it only covers the resource definition schema and the relation with [resources specifications](https://valueflows.gitbook.io/valueflows/concepts/resources#the-difference-between-a-resource-specification-and-a-resource-classification). 

IMHO in a traditional REA model we map events, an event can create and modify a resource. However, there is an open discussion in the Value Flows community about it, not only to create but also to updates resources. The overall idea is you might skip events when the actions are not economic.

https://lab.allmende.io/valueflows/valueflows/-/issues/637

**TL;DR** Weloop currently only supports creation of resources via events. However, it allows minimal resource updates without the need of events for correcting data entry errors. i.e. fixing a typo

**Comment by sbocconi** [comment]
> I don't think it answers the questions as it only covers the resource definition schema and the relation with [resources specifications](https://valueflows.gitbook.io/valueflows/concepts/resources#the-difference-between-a-resource-specification-and-a-resource-classification).

If you want to see an example using resources and events, look at this: https://valueflo.ws/examples/ex-production.html#manufacturing. But there are many on the website.

I understand Weloop only supports creation of resources via events, and this can be OK, as long as it is clear that it is Reflow's choice and not dictated by Valueflows IMHO.

**Comment by sbocconi** [comment]
Going back to this issue, I decided to ask the Value Flows community, here is the conversation:

`Stefano Bocconi @sbocconi 10:38`
Hello again, just a small question, there has been a discussion in the Reflow project about the fact that resources can enter a flow only with an event. I thought that resources could be defined independently from an event and then used in events whenever there is a change. The former interpretation requires always to choose an action appropriate for a resource when we need to describe a resource, but sometimes the resource is the boundary you want to start from.

`Mayel @mayel:matrix.org [m] 10:48`
In that case you can use a "raise" event
Of course in a well oiled circular economy you shouldn't need it too much

`Stefano Bocconi @sbocconi 10:51`
But I see in the examples that resources are just introduced without events: https://valueflo.ws/examples/ex-production.html#manufacturing

`M. @mayel:matrix.org [m] 11:24`
I'll let @fosterlynn or @bhaugen answer that

`Bob Haugen @bhaugen 12:32`
>But I see in the examples that resources are just introduced without events: https://valueflo.ws/examples/ex-production.html#manufacturing

Good catch, @sbocconi - that example does not show the events that introduced the resources. But according to REA rules, all resource changes must be made by events, so you should be able to create the current state of all resources by summarizing their events.

If you are starting a system with some resources that already exist, use an EconomicEvent with the Action "raise" to the quantity you have on hand.

Does that make sense?

(And we should probably change that example for keen-eyed observers...)

Oh wait, @mayel already mentioned "raise"....
>In that case you can use a "raise" event

I just an email notification of @sbocconi 's question and @mayel suggesting we answer and leapt to answer without reading his previous comments. We're heading out for a day trip...

**Comment by sbocconi** [comment]
So the correct action for an event creating a resource seems to be `raise`.

Is this possible in WeLoop/ReflowOS?

**Comment by viktorsmari** [comment]
I think we started with implementing only 3 actions, `consume`, `produce` and `transfer` out of total of 16 actions

![2021-08-11_14-18-48](https://user-images.githubusercontent.com/1689020/129029111-9263fac9-ede3-4d58-b4ea-47cc5a2ff9e8.png)

**Comment by sbocconi** [comment]
OK when is this (and the rest of the API) scheduled to be implemented if you know?

**Comment by viktorsmari** [comment]
@sbocconi We won't know until Guillem to returns from his vacation, about 2 weeks from now

**Comment by bhaugen** [comment]
When some software first gets implemented in a situation, often some
resources will already exist. So it's useful to have an event to enter them
into the system for the first time.

Later, sometimes mistakes will happen. I know, I know, who ever heard of
such a thing? But in the unlikely chance that they do, you might need to
correct some inventory counts. Raise increases the quantity, Lower
decreases the quantity.

**Comment by pless84** [comment]
> 
> 
> @sbocconi We won't know until Guillem to returns from his vacation, about 2 weeks from now

Is there an update on the schedule?
